cli/src/commands/examples.rs

```diff
@@ -0,0 +1,234 @@
+use octocrab::repos::RepoHandler;
+use std::collections::HashMap;
+use std::fmt;
+use std::path::{Path, PathBuf};
+use tokio::fs::File;
+use tokio::io::AsyncWriteExt;
+
+#[derive(Run, Parser, Collect, Clone)]
+#[cling(run = "run_examples")]
+pub struct Examples {
+    /// Example name
+    name: Option<String>,
+}
+
+pub async fn run_examples(example_opts: &Examples) -> Result<()> {
+    // Get latest release of the examples repo
+    let octocrab = octocrab::instance();
+    let examples_repo = octocrab.repos("restatedev", "examples");
+    let latest_release = examples_repo
+        .releases()
+        .get_latest()
+        .await
+        .context("Can't access the examples releases. Please retry later")?;
+
+    let example_asset = if let Some(example) = &example_opts.name {
+        // Check if the example exists
+        let example_lowercase = example.to_lowercase();
+        latest_release
+            .assets
+            .into_iter()
+            .find(|a| a.name.trim_end_matches(".zip") == example_lowercase)
+            .ok_or(anyhow!(
+                "Unknown example {}. Use `restate examples` to navigate the list of examples.",
+                example_lowercase
+            ))?
+    } else {
+        // Ask the example to download among the available examples
+        let mut languages = parse_available_examples(latest_release.assets);
+        let language_selection = Select::new()
+            .with_prompt("Which language?")
+            .items(&languages)
+            .interact()
+            .unwrap();
+        let example_selection = Select::new()
+            .with_prompt("Which example?")
+            .items(&languages[language_selection].examples)
+            .interact()
+            .unwrap();
+
+        languages
+            .remove(language_selection)
+            .examples
+            .remove(example_selection)
+            .asset
+    };
+
+    download_example(examples_repo, example_asset).await
+}
+
+struct Language {
+    display_name: String,
+    examples: Vec<Example>,
+}
+
+impl fmt::Display for Language {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.display_name)
+    }
+}
+
+struct Example {
+    display_name: String,
+    asset: Asset,
+}
+
+impl fmt::Display for Example {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.display_name)
+    }
+}
+
+fn parse_available_examples(assets: Vec<Asset>) -> Vec<Language> {
+    let mut languages_map = HashMap::new();
+
+    for asset in assets {
+        // Zip names have the format [language]-[example_name].zip
+
+        let asset_name = asset.name.clone();
+        let mut split_asset_name = asset_name.splitn(2, '-');
+
+        // First part is language
+        let language = if let Some(lang) = split_asset_name.next() {
+            lang.to_string()
+        } else {
+            // Bad name, just ignore it
+            continue;
+        };
```

Comment:
Good place to use `let-else`
```suggestion
        let Some(language) = split_asset_name.next() else {
            // Bad name, just ignore it
            continue;
        };
```